### DISCUSSION TITLE
Pixmap not taking /Decode value into account when dealing with /ImageMask images

### DISCUSSION THREAD

**Discussion by chatumao** [pymupdf/PyMuPDF#1192] [Upstream Bugs]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
I encountered it on a document that had the /Decode value set to [ 1 0 ] instead of the default [ 0 1 ]. After saving the image, it was white on black, but in document it is black on white.

## To Reproduce (mandatory)
Snippet:
`pdf_file = fitz.open(self.file_path)`
`for page in pdf_file:`
`images = pdf_file.get_page_images(page)`
`for image in images:`
`pix = fitz.Pixmap(pdf_file, image[0])`
`pix.invert_irect(pix.irect)`
`pix.writePNG(image_file_path)`
The file in question is sensitive, so I'd rather not send it publicly. I can provide it by request through secure channels.
Not sure about the indentation, is there a way to make it work in this context?

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).

## Your configuration (mandatory)
 - Manjaro (up to date at time of writing), x64
 - Python 3.9.6
 - PyMuPDF 1.18.15 installed by wheel

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

**Comment by JorjMcKie** [comment]
This error is happening inside the MuPDF code - which cannot be fixed inside PyMuPDF.
Please submit a bug report to their issue system https://bugs.ghostscript.com/enter_bug.cgi.
When you do this, please make sure to include a reproducer not based on PyMuPDF code. E.g. demonstrate the issue by using their CLI tool `mutool´.

If you need to find a way out urgently, use code based on [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/examples/extract-imga.py), which uses Pillow to (hopefully) correctly deal with this.

**Comment by JorjMcKie** [comment]
> Not sure about the indentation, is there a way to make it work in this context?

Encapsulate code lines inside lines

3 `grave` characters immediately followed by string "python"
... your code snippet ...
3 `grave` characters.
